src/components/stories/Stories.jsx

```diff
@@ -1,20 +1,134 @@
 import userData from "./UserData";
-import { Link } from "react-router-dom";
-import { Row, Button } from "react-bootstrap";
+import { useState } from "react";
+import { Row, Button, Modal } from "react-bootstrap";
 import UserCard from "../layout/UserCard";
+import { useForm } from "react-hook-form";
+import { supabase } from "../../utils/SupaBaseUtils";
+import { FetchingButton } from "../layout/Buttons/FetchingButton";
+import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
 import "./Stories.css";
 export default function Stories() {
+  const [show, setShow] = useState(false);
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+  const [country, setCountry] = useState("");
+  const [region, setRegion] = useState("");
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    mode: "onChange",
+    defaultValues: {
+      title: "",
+      desc: "",
+    },
+  });
+
+  const onSubmit = async (fields) => {
+    let loading = true;
+    const { data, error } = await supabase.from("stories").insert({
+      title: fields.title,
+      description: fields.desc,
+      location: region + " " + country,
+    });
+
+    while (loading) {
+      FetchingButton(loading, "btn btn-primary");
+      if (error) {
+        loading = false;
+        alert("Can not create the story! Please try again");
+      } else if (data) {
+        // clear data
+        loading = false;
+        reset({
+          title: "",
+          desc: "",
+        });
+        setCountry("");
+        setRegion("");
+        // close modal
+        handleClose();
+      }
+    }
+  };
+
   return (
     <div className="stories">
       <div className="main_title">
         <h1>User Story</h1>
-        <Button className="story_btn" as={Link} to={"/404/"}>
+        <Button className="story_btn" onClick={handleShow}>
           Create your Story
         </Button>
       </div>
       <Row className="mb-2 w-100 petList">
         {userData && userData.map((u) => <UserCard key={u.id} userData={u} />)}
       </Row>
+
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Create Your Story</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="formContainer">
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <label>Title</label>
+              <input {...register("title", { required: true })} />
```

Comment:
you could use the components from `react-bootstrap` [react-bootstrap](https://react-bootstrap.github.io/components/forms/) to make it easier
 
 add the ids to the label and input it corresponds to so we can do some testing later

Reply:
`react-bootstrap` makes it easier with the groups

```
  <Form.Group controlId="formFile" className="mb-3">
    <Form.Label>Default file input example</Form.Label>
    <Form.Control type="file" />
  </Form.Group>
  ```